### PR TITLE
DOCS: update release notes and bump version

### PR DESCRIPTION
**Pull request by SamSaffron**

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,6 @@
-Pending:
+2022-31-01 - 1.2.0
 
+- Ruby 2.6 is EOL support removed
```
Comment (tgxworld) [inline CHANGELOG:3]: It looks like we should remove Ruby 2.6 from our Ruby test matrix as part of this change: https://github.com/discourse/mini_sql/blob/bd46c55f1b9600f58e015c7720d19a6a8ffcbea3/.github/workflows/ci.yml#L40

File: lib/mini_sql/version.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # frozen_string_literal: true
 module MiniSql
-  VERSION = "1.1.3"
+  VERSION = "1.2.0"
```
Comment (tgxworld) [inline lib/mini_sql/version.rb:3]: The build is failing on an error in the gem so I'm not sure if we should do a version bump until we can be sure that the gem itself isn't broken?

Comment (SamSaffron) [reply lib/mini_sql/version.rb:3]: yes... on it ... it seems I merged a problem PR a while back it forgot to require 'set'

**Comment by SamSaffron** [comment]
@tgxworld should be fixed now.

**Comment by tgxworld** [comment]
The tests are still broken :grin:. In Ruby 3.1 and Ruby-head it looks like rubocop error is causing it. In 3.0 though, it likes we may have a leak? Unlikely to be related to this PR though but I think a green build would be great before we do a minor version bump.

**Review by tgxworld — APPROVED** [review]
The changes here looks good so approving first :+1:

**Pull request merged**